packages/drivers/odsp-driver/src/odspDriverUrlResolver2.ts

```diff
@@ -104,11 +104,10 @@ export class OdspDriverUrlResolver2 implements IUrlResolver {
 
         const odspResolvedUrl = await new OdspDriverUrlResolver().resolve(requestToBeResolved);
 
-        // Generate sharingLink only if main url is not sharing link.
-        if (!isSharingLink) {
-            try {
-                sharingLinkP = this.getShareLinkPromise(odspResolvedUrl);
-            } catch (error) {}
+        // Kick start the sharing link request if we don't already have it already as a performance optimization.
+        // For detached create new, we don't have an item id yet and therefore cannot generate a share link
+        if (!isSharingLink && odspResolvedUrl.itemId) {
+            sharingLinkP = this.getShareLinkPromise(odspResolvedUrl);
```

Comment:
@jatgarg / @leeviana  - This code is not equivalent to the old code. The way this promise is kicked off early is not the same as the version in the internal repo. With this implementation "getAbsoluteUrl" does not use a cached promise and kicks off a new network request.

Reply:
Yeah, actually we removed the sharing link cache because we started storing it in resolved url. That said, this looks fine to me. I think you should check for create new case from headers rather than checking item id:
if (request.headers && request.headers[DriverHeader.createNew]) -> Don't genrate sharing link.

Reply:
getShareLinkPromise will throw an error if itemId is undefined, not sure if createNew is the right check to do. Is this the only case itemId is undefined?

Reply:
Actually, for safety, I'm not going to change the check here. I've already tested the .itemId check so if we want to change it to be a createnew check, we should do that in a follow up issue PR.

Reply:
Yes create new is the case where item id is not there. ok, I will do that in a follow up PR. Thanks :)

Reply:
Thanks! Would be good to follow up with someone like @aboktor to make sure there aren't other cases in the create new path where item id is not set, yet we don't have the create new header set. That's my main fear with avoiding the coupling of that check to this PR

Reply:
AFAIK there isn't. This is under the control of the resolver's createCreateNewRequest though